### PR TITLE
storage: Remove spurious timestamp modification

### PR DESCRIPTION
**Pull request by tbg**
This was only used in tests and in some code paths that bypassed `Store`.

---

This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/5652)

### CODE REVIEW

File: roachpb/batch.go [hunk 1]
```diff
@@ -22,6 +22,28 @@ import (
...
 )
 
+// SetActiveTimestamp sets the correct timestamp at which the request is to be
+// carried out. For transactional requests, ba.Timestamp must be zero initially
+// and it will be set to txn.OrigTimestamp. For non-transactional requests, if
+// no timestamp is specified, nowFn is used to create and set one.
+func (ba *BatchRequest) SetActiveTimestamp(nowFn func() Timestamp) error {
+	if ba.Txn == nil {
+		// When not transactional, allow empty timestamp and simply use
+		// local clock.
```
Comment (spencerkimball) [inline roachpb/batch.go]: s/local clock/time supplied by nowFn/

File: storage/store.go [hunk 2]
```diff
@@ -1547,6 +1536,9 @@ func (s *Store) Send(ctx context.Context, ba roachpb.BatchRequest) (br *roachpb.
 	now := s.ctx.Clock.Update(ba.Timestamp)
 
 	defer func() {
+		if r := recover(); r != nil {
```
Comment (spencerkimball) [inline storage/store.go]: What's up with this?

**Comment by spencerkimball** [comment]
This looks like it moves us closer to removing that TODO for Timestamp wrangling in `Replica.beginCmds`..?

LGTM

**Comment by tamird** [comment]
Reviewed 8 of 8 files at r1.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[storage/replica_test.go, line 199 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5652#-KE1qN7S4dBEHdD0u6oh:-KE1qN7T1sx6qi17f5g6:316650303) ([raw file](https://github.com/cockroachdb/cockroach/blob/e378c08a456804a602abe38ecf7603067465cc87/storage/replica_test.go#L199)):*
nit: can this be a t.Fatal? I realize it might require some plumbing.

---

*[storage/store.go, line 1539 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5652#-KDzIgUShOMp1Z1wEtuI-r1-1539:-KE-2V3ujmzSuTu8ElDg:-658993967) ([raw file](https://github.com/cockroachdb/cockroach/blob/e378c08a456804a602abe38ecf7603067465cc87/storage/store.go#L1539)):*
I guess it's here to avoid running this `defer` in case of a panic. Probably should have a comment to explain why.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/5652)*

**Comment by tschottdorf** [comment]
Review status: 3 of 8 files reviewed at latest revision, 3 unresolved discussions, some commit checks failed.

---

*[roachpb/batch.go, line 32 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5652#-KDzIgUShOMp1Z1wEtuC-r1-32:-KE7qOt8ftqig08eTay0:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/e378c08a456804a602abe38ecf7603067465cc87/roachpb/batch.go#L32)):*
Done.

---

*[storage/replica_test.go, line 199 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5652#-KE1qN7S4dBEHdD0u6oh:-KE7bf3HTGcwf_FIjgN4:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/e378c08a456804a602abe38ecf7603067465cc87/storage/replica_test.go#L199)):*
Done.

---

*[storage/store.go, line 1539 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/5652#-KDzIgUShOMp1Z1wEtuI-r1-1539:-KE7qMJFdqA7ZLIc62Bj:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/e378c08a456804a602abe38ecf7603067465cc87/storage/store.go#L1539)):*
Done.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/5652)*

**Comment by tamird** [comment]
Reviewed 5 of 5 files at r2.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/cockroachdb/cockroach/5652)*

**Pull request merged**